lib/asyn-thread.c

```diff
@@ -298,6 +326,15 @@ static unsigned int CURL_STDCALL getaddrinfo_thread(void *arg)
     free(td);
   }
   else {
+#ifdef HAVE_SOCKETPAIR
+    if(tsd->sock_pair[1] != INVALID_SOCKET) {
+    /* DNS has been resolved, signal client task */
+      if(write(tsd->sock_pair[1],  buf, sizeof(buf)) < 0) {
+        /* update sock_erro to errno */
+        tsd->sock_error = SOCKERRNO;
```

Comment:
If this happens, we're in deep trouble since now the main thread won't know the thread is done. Can do anything more than this?

Reply:
Main thread is polling for td->tsd.done flag and since we are setting this flag even in case of write error and would result into DNS resolve failure.

CURLcode Curl_resolver_is_resolved(struct connectdata *conn,
                                   struct Curl_dns_entry **entry) {
  Curl_mutex_acquire(td->tsd.mtx);
  done = td->tsd.done;
  Curl_mutex_release(td->tsd.mtx);

  if(done) {
    getaddrinfo_complete(conn);

    if(!conn->async.dns) {
      CURLcode result = resolver_error(conn);
      destroy_async_data(&conn->async);
      return result;
    }
    destroy_async_data(&conn->async);
    *entry = conn->async.dns;
  }



Another option can be not to set below mentioned error status, client will not be notified and let connection request gets delay due to application polling ?. With this approach at-least request would get processed. 
tsd->sock_error = SOCKERRNO;

Please provide your opinion.